C#/Botkit/Microsoft.BotKit.Adapters.Slack/SlackBotWorker.cs

```diff
@@ -135,16 +279,24 @@ public async Task<object> ReplyWithDialog(object source, object dialogObj)
         /// <param name="update">An object in the form `{id: <id of message to update>, conversation: { id: <channel> }, text: <new text>, card: <array of card objects>}`</param>
         public async Task<object> UpdateMessage(IBotkitMessage update)
         {
-            return new object();
+            SlackAdapter adapter = (SlackAdapter)GetController().Adapter;
+            TurnContext context = (TurnContext)GetConfig("context");
+            CancellationTokenSource source = new CancellationTokenSource();
+            CancellationToken token = source.Token;
+            return await adapter.UpdateActivityAsync(context, update.IncomingMessage, token);
         }
 
         /// <summary>
         /// Delete an existing message.
         /// </summary>
         /// <param name="update">An object in the form of `{id: <id of message to delete>, conversation: { id: <channel of message> }}`</param>
-        public async Task<object> DeleteMessage(IBotkitMessage update)
+        public async Task DeleteMessage(IBotkitMessage update)
```

Comment:
I believe this should return a `Task<DeletedResponse>`

Reply:
I don't know about that, the method DeleteActivityAsync returns void